common/reactive-grpc-common/src/main/java/com/salesforce/reactivegrpc/common/AbstractStreamObserverAndPublisher.java

```diff
@@ -354,6 +367,30 @@ public void request(long n) {
         }
     }
 
+    /**
+     * Concurrent addition bound to Long.MAX_VALUE.
+     * Any concurrent write will "happen before" this operation.
+     *
+     * @param <T> the parent instance type
+     * @param updater  current field updater
+     * @param instance current instance to update
+     * @param toAdd    delta to add
+     * @return value before addition or Long.MAX_VALUE
+     */
+    private static <T> long addCap(AtomicLongFieldUpdater<T> updater, T instance, long toAdd) {
+        long r, u;
+        for (;;) {
+            r = updater.get(instance);
+            if (r == Long.MAX_VALUE) {
+                return Long.MAX_VALUE;
+            }
+            u = LongMath.saturatedAdd(r, toAdd);
```

Comment:
@rmichela I think this will not work since in case of overflow we want to get Long.MAX_VALUE

```
  /**
   * Returns the sum of {@code a} and {@code b} unless it would overflow or underflow in which case
   * {@code Long.MAX_VALUE} or {@code Long.MIN_VALUE} is returned, respectively.
   *
   * @since 20.0
   */
  @Beta
  public static long saturatedAdd(long a, long b) {
    long naiveSum = a + b;
    if ((a ^ b) < 0 | (a ^ naiveSum) >= 0) {
      // If a and b have different signs or a has the same sign as the result then there was no
      // overflow, return.
      return naiveSum;
    }
    // we did over/under flow, if the sign is negative we should return MAX otherwise MIN
    return Long.MAX_VALUE + ((naiveSum >>> (Long.SIZE - 1)) ^ 1);
  }
```

Reply:
Overflow returns Long.MAX_VALUE, as I understand it.